BE/src/main/java/louie/dong/airbnb/accommodation/AccommodationService.java

```diff
@@ -26,91 +27,63 @@
 @Service
 @RequiredArgsConstructor
 @Transactional(readOnly = true)
+@Slf4j
 public class AccommodationService {
 
-	private final AccommodationRepository accommodationRepository;
-	private final WishRepository wishRepository;
-
-	public AccommodationPriceResponse findPrices(String country) {
-		List<Integer> prices = accommodationRepository.findByAccommodationPrices(country);
-		return new AccommodationPriceResponse(calculateAverage(prices), prices);
-	}
-
-	public AccommodationDetailResponse findById(Long id) {
-		Accommodation accommodation = getAccommodation(id);
-		boolean wish = wishRepository.existsByAccommodationId(id);
-		return new AccommodationDetailResponse(accommodation, wish);
-	}
-
-	public AccommodationDetailPriceResponse findDetailPrice(Long id,
-		AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
-		Accommodation accommodation = getAccommodation(id);
-		LocalDate checkIn = accommodationDetailPriceRequest.getCheckIn();
-		LocalDate checkOut = accommodationDetailPriceRequest.getCheckOut();
-
-		int date = (int) checkIn.until(checkOut, ChronoUnit.DAYS);
-		int totalPrice = accommodation.getPrice() * date;
-		double discountRate = getDiscountRate(date) * 0.01;
-		int discountPrice = (int) (totalPrice * discountRate);
-		int finalPrice = totalPrice - discountPrice + accommodation.getCleaningFee()
-			+ accommodation.getServiceFee() + accommodation.getAccommodationFee();
-
-		return new AccommodationDetailPriceResponse(accommodation.getPrice(), date, totalPrice,
-			WEEKLY.getDiscountRate(), discountPrice, accommodation.getCleaningFee(),
-			accommodation.getServiceFee(), accommodation.getAccommodationFee(), finalPrice);
-	}
-
-	public AccommodationSearchResponse findAccommodations(
-		AccommodationSearchRequest accommodationSearchRequest) {
+    private final AccommodationRepository accommodationRepository;
+    private final WishRepository wishRepository;
+
+    public AccommodationPriceResponse findPrices(String country) {
+        List<Integer> prices = accommodationRepository.findPricesByAccommodation(country);
+        return new AccommodationPriceResponse(calculateAverage(prices), prices);
+    }
+
+    public AccommodationDetailResponse findById(Long id) {
+        Accommodation accommodation = getAccommodationOrThrow(id);
+        boolean wish = wishRepository.existsByAccommodationId(id);
+        return new AccommodationDetailResponse(accommodation, wish);
+    }
+
+    public AccommodationDetailPriceResponse findDetailPrice(Long id,
+        AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
+        Accommodation accommodation = getAccommodationOrThrow(id);
+        LocalDate checkIn = accommodationDetailPriceRequest.getCheckIn();
+        LocalDate checkOut = accommodationDetailPriceRequest.getCheckOut();
+        return new AccommodationDetailPriceResponse(checkIn, checkOut, accommodation);
+    }
```

Comment:
💯

Reply:
잘 적용해 주셔서 한발 더 나가보자면..
도메인 객체를 좀 더 나눈다면 아래와 같이도 사용 할 수 있을것 같아요

```suggestion
    public AccommodationDetailPriceResponse findDetailPrice(Long id,
        AccommodationDetailPriceRequest accommodationDetailPriceRequest) {
        Accommodation accommodation = getAccommodationOrThrow(id);
        Period period = accommodationDetailPriceRequest.toPeriod();
        return new AccommodationDetailPriceResponse(period, accommodation);
    }
    //...
    
    class Period {
       private LocalDate checkIn;
       private LocalDate checkOut;
       //....
```

타 도메인 객체가 가지고 있는 기간에 대한 책임을 Period(가명) 라는 객체가 가질수 있을것 같네요(단일책임 원칙 | [SOLID](https://ko.wikipedia.org/wiki/SOLID_(%EA%B0%9D%EC%B2%B4_%EC%A7%80%ED%96%A5_%EC%84%A4%EA%B3%84)))